app/controllers/two_factor_authentication/personal_key_verification_controller.rb

```diff
@@ -66,8 +66,7 @@ def alert_user_about_personal_key_sign_in(disavowal_token)
 
     def remove_personal_key
       # for now we will regenerate a key and not show it to them so retire personal key page shows
-      current_user.personal_key = PersonalKeyGenerator.new(current_user).create
-      current_user.save!
+      PersonalKeyGenerator.new(current_user).create
```

Comment:
would it be clearer if we flipped this? like removed the `user.personal_key = ` assignment from from inside the `PersonalKeyGenerator`?

Or maybe we rename the method from `#create` to like `#assign!` so it's clear there's that side effect?

Reply:
The hurdle is there are a handful of tests that rely on the behavior ([example](https://github.com/18F/identity-idp/blob/7753d43b55505a2ad79d707210e58810ae5bb0d0/spec/features/two_factor_authentication/sign_in_via_personal_key_spec.rb#L9)).

`#create` definitely doesn't tell the whole story, so a rename makes sense to me. Something like `encrypt_and_save!`?

Reply:
It's just tests? They already have the user handy so they could do:

```ruby
user.personal_key = PersonalKeyGenerator.new(...).create
user.save!
```

Reply:
It was closer to 20 and I think I lean towards keeping that behavior in the class. If we did remove the assignment/save, we'd probably want to refactor a bit more since the user being passed in would only be used for personal key verification (which maybe shouldn't be in this class based on the name?).

I've renamed the method `#generate!` to try to make the existence of side effects clearer.